### DISCUSSION TITLE
Automatically get image resolution for next/image at build-time

### DISCUSSION THREAD

**Discussion by danptr** [vercel/next.js#18863] [Help]
I have a blog that used MDX. I'd like to start using `next/image` for all the images in my articles. However, I do NOT want to go through every <blogpost>.mdx and change every <img> tag to a `next/image` component and look up the resolution of every image file. Since everything happens on the server at build-time there is no reason why I shouldn't be able to do this programatically, right?

What I have in mind:

- Using `getStaticProps` to fetch the image res and then inject this into my image component inside `MDXProvider`
   -  I don't know if that's possible since `getStaticProps` only works with pages
- Writing a custom loader that extends `file-loader` and fetch the metadata there

Has anyone done anything similar and/or can give me some advice on this? Ideally I'd like to achieve this within Next.js without messing around with Webpack.

**Comment by lfades** [comment]
> Since everything happens on the server at build-time there is no reason why I shouldn't be able to do this programatically, right?

That's not how it works, it happens at runtime and `next/image` is required atm.

Using MDX you can change the component used for images and include `next/image` there, updating all of your images in the process 👍

**Reply by danptr** [reply]
> Using MDX you can change the component used for images and include next/image there, updating all of your images in the process

Yes, that's what I have in mind. But the MDX to React component mapper is just an object that contains React components for each tag. How can I hook up `getStaticProps` to that so I can run Node code to fetch the image resolution?

**Comment by danptr** [comment]
In case I wasn't clear enough before, the issue I face is wanting to run server-side code somewhere in the MDX process. Let's take @leerob leerob.io [blog](https://github.com/leerob/leerob.io) as an example, since it's a wonderful case for MDX in Next.js. If I were to convert all the blog images to `next/image` without changing each article manually and adding the width and height by hand, I'd ideally do this in ``:

```javascript
// /components/MDXComponents.js

const Image = ({ src, alt }) => {
  // Call Node-only function that fetches image width / height on filesystem
  const { width, height } = getImageResolution(src)

  return <NextImage src={url} width={width} height={height} alt={alt} />
}

const MDXComponents = {
  h1: (props) => <Heading as="h1" size="xl" my={4} {...props} />,
  img: Image
  [...]
};

export default MDXComponents
```
Obviously I cannot call a Node function (which needs `fs`) in a React component, even if this component will only ever be prerendered. But I'd be great if I could use `getStaticProps` or something similar to achieve this.
